### PR TITLE
fixes #85 and DRYs out versioning

### LINKED ISSUE #85: Asset.updated() crashes every time
Looks like a capitalization problem.  I think it should be `raise NotImplementedError('...')` though since apparently the subclasses have to do it.

https://github.com/Applied-GeoSolutions/gips/blob/3ac0bd80ef7360a10ccbbf542b07bc9590604b31/gips/data/core.py#L219

### PR DESCRIPTION
**Pull request by ircwaves**
Now drivers only need to set `self._version` to get the default version handling behavior.

Example for possible over-ride might be to consider `C1` and `DN` equivalent in the landsat driver.

### CODE REVIEW

File: gips/data/core.py [hunk 1]
```diff
@@ -255,13 +255,24 @@ def __init__(self, filename):
...
         '''
-        Compare the version info for this asset (self) to that of newasset.
-        Return true if newasset version is greater.
+        Return:
+            'newasset' and existing represent the same data (time,space,sensor)
+            AND
+            'newasset' _version greater than existing _version.
+
         '''
-        return false
```
Comment (ags-tolson) [inline gips/data/core.py]: Today is a good day.

File: gips/data/core.py [hunk 2]
```diff
@@ -563,18 +574,21 @@ def _archivefile(cls, filename, update=False):
...
                     otherversions = True
                 elif len(existing) > 0 and update:
-                    # update case:  Remove existing outdated assets and install the new one
+                    # update case:  Remove existing outdated assets
+                    #               and install the new one
                     VerboseOut('%s: removing other version(s):' % bname, 1)
                     for ef in existing:
-                        assert ef.updated(asset), 'Asset is not updated version'
+                        if not ef.updated(asset):
+                            'Asset is not updated version'
```
Comment (ags-tolson) [inline gips/data/core.py]: Is this line missing a call or an assignment to go with the string?

File: gips/data/prism/prism.py [hunk 3]
```diff
@@ -156,21 +156,10 @@ def __init__(self, filename):
         self.scale = scale
         self.version = version
         self.stability = stability
-        self.ver_stab = self._stab_score[self.stability] * .01 + int(self.version[1:])
+        self._version = self._stab_score[self.stability] * .01 + int(self.version[1:])
```
Comment (ags-tolson) [inline gips/data/prism/prism.py]: Looks like `self.ver_stab` is used in `PrismAsset.query_provider`; I think it just needs to be changed to `self._version`.

File: gips/data/merra/merra.py [hunk 4]
```diff
@@ -259,16 +259,6 @@ def fetch(cls, asset, tile, date):
...
-    def updated(self, newasset):
-        '''
-        Compare the version for this to that of newasset.
-        Return true if newasset version is greater.
-        '''
-        return (self.sensor == newasset.sensor and
-                self.tile == newasset.tile and
-                self.date == newasset.date and
-                self.version < newasset.version)
-
```
Comment (ags-tolson) [inline gips/data/merra/merra.py]: Double kill!

File: gips/data/modis/modis.py [hunk 5]
```diff
@@ -251,17 +251,6 @@ def fetch(cls, asset, tile, date):
...
-        '''
-        Compare the version for this to that of newasset.
-        Return true if newasset version is greater.
-        '''
-        return (self.sensor == newasset.sensor and
-                self.tile == newasset.tile and
-                self.date == newasset.date and
-                self.version < newasset.version)
-
-
```
Comment (ags-tolson) [inline gips/data/modis/modis.py]: Triple Kill!

File: gips/data/prism/prism.py [hunk 6]
```diff
@@ -156,21 +156,10 @@ def __init__(self, filename):
...
-        '''
-        Compare the version for this to that of newasset.
-        Return true if newasset version is greater.
-        '''
-        return (self.sensor == newasset.sensor and
-                self.tile == newasset.tile and
-                self.date == newasset.date and
-                self.asset == self.asset and
-                self.ver_stab < newasset.ver_stab)
-
```
Comment (ags-tolson) [inline gips/data/prism/prism.py]: Quadruple Kill!

File: gips/data/sentinel2/sentinel2.py [hunk 7]
```diff
@@ -428,17 +431,6 @@ def tile_list(cls, file_name):
...
-        '''
-        Compare the version for this to that of newasset.
-        Return true if newasset version is greater.
-        '''
-        return (self.sensor == newasset.sensor and
-                self.tile == newasset.tile and
-                self.date == newasset.date and
-                self.version < newasset.version)
-
-
```
Comment (ags-tolson) [inline gips/data/sentinel2/sentinel2.py]: QUINTUPLE KILL!  https://www.youtube.com/watch?v=kqCvEAiOVGQ

File: gips/data/core.py (hunk 2 above)
Comment (ircwaves) [reply gips/data/core.py]: Now that is getting distracted at an inopportune, but syntactically correct time.  Was going to replace the assertionerror, but hadn't figured it, and got pulled into something else.

File: gips/data/prism/prism.py (hunk 3 above)
Comment (ircwaves) [reply gips/data/prism/prism.py]: Good catch.

Comment (ags-tolson) [reply gips/data/prism/prism.py]: resolved by 9e4230d

File: gips/data/core.py [hunk 8]
```diff
@@ -561,20 +572,33 @@ def _archivefile(cls, filename, update=False):
...
                     for ef in existing:
-                        assert ef.updated(asset), 'Asset is not updated version'
+                        if not ef.updated(asset):
+                            utils.verbose_out(
+                                'Asset {} is not updated version of {}.'
+                                .format(ef.filename, asset.filename) +
+                                ' Remove existing asset to replace.', 2
+                            )
+                            # NOTE: This return makes sense iff len(existing)
+                            # cannot be greater than 1
```
Comment (ags-tolson) [inline gips/data/core.py]: Optional:  Maybe put an assertion or warning just before the loop, to confirm that `len(existing)` has a sane value?

Comment (ircwaves) [reply gips/data/core.py]: This note was meant more as something to key in on iff strange behavior occurs.  I guess the assert would make that even more plain and get to the root of things quicker.

**Pull request merged**